File: Java/백준/Silver/1972. 놀라운 문자열/놀라운 문자열.java
import java.util.*;
import java.io.*;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static String word, str;
    static boolean flag;
    static int total;
    static Set<String> s = new HashSet<>();

    public static void main(String[] args) throws IOException {


        while (true) {
            word = br.readLine();
            if (word.equals("*"))
                break;

            flag = true;
            for (int i = 1; i < word.length(); i++) {
                total = 0;
                s.clear();
                for (int j = 0; j < word.length() - i; j++) {
                    str = String.valueOf(word.charAt(j)) + String.valueOf(word.charAt(j + i));
                    s.add(str);
                    total++;
                }

                if (s.size() != total) {
                    flag = false;
                    break;
                }
            }

            if (flag)
                bw.write(String.format("%s is surprising.\n", word));
            else
                bw.write(String.format("%s is NOT surprising.\n", word));
        }

        bw.close();
    }
}

File: PyPy3/백준/Silver/1972. 놀라운 문자열/놀라운 문자열.py
from sys import stdin

input = stdin.readline

while True:
    word = input().strip()
    if word == "*":
        break

    flag = True  # 중복 여부 변수 
    for i in range(1, len(word)):
        s = set()
        total = 0
        for j in range(len(word) - i):
            st = word[j] + word[j + i]
            s.add(st)
            total += 1

        # 중복되는 경우가 있다면 -> 개수가 틀리다는 점을 이용
        if total != len(s):
            flag = False
            break

    if flag:
        print(f"{word} is surprising.")
    else:
        print(f"{word} is NOT surprising.")
